services/director-v2/src/simcore_service_director_v2/modules/db/repositories/comp_pipelines.py

```diff
@@ -16,14 +16,14 @@
 
 class CompPipelinesRepository(BaseRepository):
     @log_decorator(logger=logger)
-    async def publish_pipeline(
-        self, project_id: ProjectID, dag_graph: nx.DiGraph
+    async def upsert_pipeline(
```

Comment:
And if you really want to get 100% readable I'd use `create_or_update_pipeline` like some ORMs do. 😄

Reply:
well actually upsert is a defined name, meaning update or insert